x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/query/ScoreScriptUtilsTests.java

```diff
@@ -128,21 +128,58 @@ public void testSparseVectorFunctionsSpecialCases() {
         // test dotProduct
         DotProductSparse docProductSparse = new DotProductSparse(queryVector);
         double result = docProductSparse.dotProductSparse(dvs);
-        assertEquals("dotProductSparse result is not equal to the expected value!", 65425.62, result, 0.1);
+        assertEquals("dotProductSparse result is not equal to the expected value!", 65425.626, result, 0.001);
 
         // test cosineSimilarity
         CosineSimilaritySparse cosineSimilaritySparse = new CosineSimilaritySparse(queryVector);
         double result2 = cosineSimilaritySparse.cosineSimilaritySparse(dvs);
-        assertEquals("cosineSimilaritySparse result is not equal to the expected value!", 0.78, result2, 0.1);
+        assertEquals("cosineSimilaritySparse result is not equal to the expected value!", 0.786, result2, 0.001);
 
         // test l1norm
         L1NormSparse l1Norm = new L1NormSparse(queryVector);
         double result3 = l1Norm.l1normSparse(dvs);
-        assertEquals("l1normSparse result is not equal to the expected value!", 517.18, result3, 0.1);
+        assertEquals("l1normSparse result is not equal to the expected value!", 517.184, result3, 0.001);
```

Comment:
I would have expeced the l1 norm to change by 11.5 when adding the "4546" doc vector dimentions of 11.5f? 
Could you re-check the expected value? I think the while-loop in the function might miss the last "biggest" value?
Same for the l2-norm also btw.

Reply:
Sorry, my bad. I now saw you took out the "11" and moved it to "4546" with the same value. All good.